templates/docs/contribute/design-proposal.md

```diff
@@ -0,0 +1,102 @@
+2. Create a share link for your file and set permissions to “anyone can view”
+3. Add the link to the GitHub issue
+
+The Figma library maintainers at Canonical will then review your work and add it into the shared library
+
+At Canonical, we currently use Figma for our design libraries, but you can also create the design for your proposal with a different tool, such as [Penpot](https://penpot.app/).
+
+## Document your proposal
+
+Documentation works as a ‘user manual’ for your component or pattern. It's your chance to tell users how they should use it. We always aim for high-quality, detailed documentation.
+
+To get started with the minimal documentation, answer the following questions in clear, plain language:
+
+| Questions:                           | Answers:                                     | For example:                                                                                                                                                                                                                                                  |
+| :----------------------------------- | :------------------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| What are the parts of the component? | Parts, or ‘anatomy’                          | List of component parts and their names Illustrations to show structure                                                                                                                                                                                       |
+|                                      | Variants                                     | Different implementations of the component Examples like blog cards versus Charmhub cards                                                                                                                                                                     |
+|                                      | Properties                                   | Configurable properties like size and color Show/hide options Editable text fields                                                                                                                                                                            |
+| When should we use the component?    | When to use (best practices)                 | Context and problem-solving scenarios Alternative component options Comparative advantages UI examples showing correct implementation Recommended practices or do’s and dont’s Usage limitations (such as primary button frequency) Implementation guidelines |
```

Comment:
"do's and dont's" is currently causing a [spellcheck error](https://github.com/canonical/vanilla-framework/actions/runs/17241928642/job/48921252408?pr=5626).

We can add a spelling exception to allow this, but we have a choice ahead of us, maybe you should check with someone from Content.

[This article](https://www.vocabulary.com/articles/wc/dos-and-donts-or-dos-and-donts/) says that some style guides recommend "dos and don'ts", others recommend "do's and don'ts.", and others recommend "do's and don't's". We need to check the style guide to see what the recommendation across the company is.